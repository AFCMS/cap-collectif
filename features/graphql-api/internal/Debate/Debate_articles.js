/* eslint-env jest */
const DebateArticlesQuery = /* GraphQL */ `
  query DebateArticlesQuery($id: ID!, $count: Int!, $cursor: String) {
    node(id: $id) {
      ... on Debate {
        articles(first: $count, after: $cursor) {
          edges {
            cursor
            node {
              id
              url
              title
              description
              coverUrl
              origin
              publishedAt
              hasBeenCrawled
            }
          }
        }
      }
    }
  }
`;

const DebateArticlesCountersQuery = /* GraphQL */ `
  query DebateArticlesCountersQuery($id: ID!) {
    node(id: $id) {
      ... on Debate {
        articles(first: 0) {
          totalCount
        }
      }
    }
  }
`;

describe('Internal|Debate.articles connection', () => {
  const debateId = toGlobalId('Debate', 'debateCannabis');

  it('fetches 2 first articles associated to a debate with a cursor', async () => {
    await expect(
      graphql(
        DebateArticlesQuery,
        {
          count: 2,
          id: debateId,
          cursor: null,
        },
        'internal',
      ),
    ).resolves.toMatchSnapshot();
  });

  it('fetches 1 next articles associated to a debate with a cursor', async () => {
    await expect(
      graphql(
        DebateArticlesQuery,
        {
          count: 1,
          id: debateId,
          cursor: toGlobalId('arrayconnection', '1'),
        },
        'internal',
      ),
    ).resolves.toMatchSnapshot();
  });

  it('fetches votes counters associated to a debate', async () => {
    await expect(
      graphql(
        DebateArticlesCountersQuery,
        {
          id: debateId,
        },
        'internal',
      ),
    ).resolves.toMatchSnapshot();
  });
});
