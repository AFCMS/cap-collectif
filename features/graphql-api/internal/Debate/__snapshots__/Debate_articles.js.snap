// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Internal|Debate.articles connection fetches 1 next articles associated to a debate with a cursor 1`] = `
Object {
  "node": Object {
    "articles": Object {
      "edges": Array [
        Object {
          "cursor": "YXJyYXljb25uZWN0aW9uOjI=",
          "node": Object {
            "coverUrl": "https://media.ouest-france.fr/v1/pictures/MjAyMDEwODBiNzFjMTIwZDFmMTEyMTBhZjY2NTJlNzVjMGM2MDk?width=1260&focuspoint=50%2C25&cropresize=1&client_id=bpeditorial&sign=1c59c4f45f057ef4aca074a7f52c1a7368ab4dd34705f990db7b2bca452d1aa5",
            "description": "Un décret publié ce vendredi autorise les premières expérimentations du cannabis
thérapeutique. Elles concerneront 3 000 patients avec des médicaments qui se
présenteront sous forme d’huiles.
",
            "hasBeenCrawled": true,
            "id": "RGViYXRlQXJ0aWNsZTpjYW5hYmlzQXJ0aWNsZU91ZXN0RnJhbmNlMg==",
            "origin": "Ouest France",
            "publishedAt": "2020-10-01 00:00:00",
            "title": "Cannabis thérapeutique. Les premières expérimentations autorisées par décret",
            "url": "https://www.ouest-france.fr/societe/cannabis/cannabis-therapeutique-les-premieres-experimentations-autorisees-par-decret-7007774",
          },
        },
      ],
    },
  },
}
`;

exports[`Internal|Debate.articles connection fetches 2 first articles associated to a debate with a cursor 1`] = `
Object {
  "node": Object {
    "articles": Object {
      "edges": Array [
        Object {
          "cursor": "YXJyYXljb25uZWN0aW9uOjA=",
          "node": Object {
            "coverUrl": "https://www.royalqueenseeds.fr/modules/prestablog/themes/royalqueenseeds/up-img/74.jpg",
            "description": "Alors que le débat sur la légalisation du cannabis fait son retour, une étude du
média spécialisé Newsweed estime que le marché noir du stupéfiant le plus consommé
de France serait bien plus important que ce qu'on estimait.
",
            "hasBeenCrawled": true,
            "id": "RGViYXRlQXJ0aWNsZTpjYW5hYmlzQXJ0aWNsZUJmbQ==",
            "origin": "BFM TV",
            "publishedAt": "2020-10-01 00:00:00",
            "title": "Que pèse vraiment le marché du cannabis en France ?",
            "url": "https://www.bfmtv.com/economie/estime-a-3-2-milliards-d-euros-en-france-le-marche-du-cannabis-serait-bien-plus-gros-que-ce-qu-on-pensait_AN-202010200002.html",
          },
        },
        Object {
          "cursor": "YXJyYXljb25uZWN0aW9uOjE=",
          "node": Object {
            "coverUrl": "https://media.ouest-france.fr/v1/pictures/MjAyMDEwOWJjYjQyNmYxMDc3ZDlmZmQ0MDlhMzY3MDk4MGI5Y2U?width=1260&focuspoint=50%2C25&cropresize=1&client_id=bpeditorial&sign=8435fcd4c3a6b0678e792827aedeb9f7447a2ee1c7ba8dc1c8944674643dda1f",
            "description": "Les futurs médicaments à base de cannabis, seront utilisés
dans le cadre d’une expérimentation à partir de l’année prochaine, en France.
",
            "hasBeenCrawled": true,
            "id": "RGViYXRlQXJ0aWNsZTpjYW5hYmlzQXJ0aWNsZU91ZXN0RnJhbmNlMQ==",
            "origin": "Ouest France",
            "publishedAt": "2020-10-01 00:00:00",
            "title": "Cannabis thérapeutique : des médicaments sous forme d’huiles, gélules et fleurs séchées expérimentés",
            "url": "https://www.ouest-france.fr/societe/cannabis/cannabis-therapeutique-le-cahier-des-charges-des-futurs-medicaments-publie-7020873",
          },
        },
      ],
    },
  },
}
`;

exports[`Internal|Debate.articles connection fetches votes counters associated to a debate 1`] = `
Object {
  "node": Object {
    "articles": Object {
      "totalCount": 3,
    },
  },
}
`;
